sensitivity-computation/src/main/java/com/farao_community/farao/sensitivity_computation/SystematicSensitivityInterface.java

```diff
@@ -0,0 +1,212 @@
+
+
+/**
+ * A computation engine dedicated to the systematic sensitivity analyses performed
+ * in the scope of the LinearRao.
+ *
+ * @author Viktor Terrier {@literal <viktor.terrier at rte-france.com>}
+ * @author Philippe Edwards {@literal <philippe.edwards at rte-france.com>}
+ * @author Baptiste Seguinot {@literal <baptiste.seguinot at rte-france.com>}
+ */
+public final class SystematicSensitivityInterface {
+
+    private static final Logger LOGGER = LoggerFactory.getLogger(SystematicSensitivityInterface.class);
+
+    /**
+     * LinearRao configurations, containing the default and fallback configurations
+     * of the sensitivity computation
+     */
+    private SensitivityComputationParameters defaultParameters;
+
+    private SensitivityComputationParameters fallbackParameters;
+
+    /**
+     * The sensitivity provider to be used in the sensitivity computation
+     */
+    private SensitivityProvider sensitivityProvider;
+
+    /**
+     * A boolean indicating whether or not the fallback mode of the sensitivity computation
+     * engine is active.
+     */
+    private boolean fallbackMode = false;
+
+    /**
+     * Constructors
+     */
+    public static final class SystematicSensitivityInterfaceBuilder {
+        private SensitivityComputationParameters defaultParameters;
+        private SensitivityComputationParameters fallbackParameters;
+        private SensitivityProvider sensitivityProvider;
+        private PtdfSensitivityProvider ptdfSensitivityProvider;
+        private RangeActionSensitivityProvider rangeActionSensitivityProvider;
+
+        private SystematicSensitivityInterfaceBuilder() {
+
+        }
+
+        public SystematicSensitivityInterfaceBuilder withFallbackParameters(SensitivityComputationParameters fallbackParameters) {
+            this.fallbackParameters = fallbackParameters;
+            return this;
+        }
+
+        public SystematicSensitivityInterfaceBuilder withDefaultParameters(SensitivityComputationParameters defaultParameters) {
+            this.defaultParameters = defaultParameters;
+            return this;
+        }
+
+        public SystematicSensitivityInterfaceBuilder withSensitivityProvider(SensitivityProvider sensitivityProvider) {
+            this.sensitivityProvider = sensitivityProvider;
+            return this;
+        }
+
+        public SystematicSensitivityInterfaceBuilder withPtdfSensitivities(GlskProvider glskProvider, Set<Cnec> cnecs) {
+            ptdfSensitivityProvider = new PtdfSensitivityProvider(glskProvider);
+            ptdfSensitivityProvider.addCnecs(cnecs);
+            return this;
+        }
+
+        public SystematicSensitivityInterfaceBuilder withRangeActionSensitivities(Set<RangeAction> rangeActions, Set<Cnec> cnecs) {
+            rangeActionSensitivityProvider = new RangeActionSensitivityProvider();
+            rangeActionSensitivityProvider.addSensitivityFactors(rangeActions, cnecs);
+            return this;
+        }
```

Comment:
What if the builder owns a MultipleSensitivityProvider created at construction, and then you just add the different providers to the already created MultipleSensitivityProvider ? Then in the build method you don't have to do all the checks ?